week5/palinl.c

```diff
@@ -0,0 +1,85 @@
+/*C Program to Check whether a Singly Linked List is a Palindrome */
+#include <stdio.h>
+#include <stdlib.h>
+ 
+struct node
+{
+    int num;
+    struct node *next;
+};
+
+ 
+int main()
+{
+    struct node *p = NULL;
+    int c, ch, count = 0,n,pc=0;
+    printf("Enter data into the list\n");
+    struct node *temp,**link=NULL;
+    n=create(&p);
+    pc = CheckPalin(p, n);
+    if (pc == 1)
+    {
+        printf("palindrome linked list\n");
+    }
+    else
+    {
+        printf("non-palindrome linked list\n");
+    }
+
+ 
+    return 0;
+}
+
+int create (struct node **link)
```

Comment:
In the real case you would be given the list, so the count n would need to be calculated. How would you do that?